### PR TITLE
expression, parser: add function octet_length support

### PR DESCRIPTION
**Pull request by gaoxingliang**
### What problem does this PR solve? 
Fix https://github.com/pingcap/tidb/issues/11195 add function octet_length support

the PR in parser is https://github.com/pingcap/parser/pull/401

### What is changed and how it works?
it's same with the length function. so just register it into the map 

### Check List 

Tests 

 - Unit test   - added UT cases

Code changes

 - Has exported function/method change  -- add octet_length support

### CODE REVIEW

**Comment by zz-jason** [comment]
@gaoxingliang Thanks for your contribution! It would be nice if you can fix the CI

**Comment by gaoxingliang** [comment]
@zz-jason I think I need to wait the PR in parser is merged and then changed the go module to the original parser repo. then the CI can pass. is that right? I build it locally. `make dev` passed.

File: expression/builtin.go [hunk 1]
```diff
@@ -458,6 +458,7 @@ var funcs = map[string]functionClass{
 	ast.Left:            &leftFunctionClass{baseFunctionClass{ast.Left, 2, 2}},
 	ast.Right:           &rightFunctionClass{baseFunctionClass{ast.Right, 2, 2}},
 	ast.Length:          &lengthFunctionClass{baseFunctionClass{ast.Length, 1, 1}},
+	ast.OctetLength:     &lengthFunctionClass{baseFunctionClass{ast.OctetLength, 1, 1}}, // synonym
```
Comment (SunRunAway) [inline expression/builtin.go]: Suggest to use a more precise comment.

```go
    // This function is a synonym for LENGTH().
	// See https://dev.mysql.com/doc/refman/5.7/en/string-functions.html#function_octet-length
	ast.OctetLength:     &lengthFunctionClass{baseFunctionClass{ast.OctetLength, 1, 1}},
```

Comment (gaoxingliang) [reply expression/builtin.go]: fixed

**Comment by gaoxingliang** [comment]
The automatically go fmt removed some blankspace for the files.

File: expression/builtin.go [hunk 2]
```diff
@@ -442,30 +442,33 @@ var funcs = map[string]functionClass{
...
+	ast.LoadFile:   &loadFileFunctionClass{baseFunctionClass{ast.LoadFile, 1, 1}},
+	ast.Locate:     &locateFunctionClass{baseFunctionClass{ast.Locate, 2, 3}},
+	ast.Lower:      &lowerFunctionClass{baseFunctionClass{ast.Lower, 1, 1}},
+	ast.Lpad:       &lpadFunctionClass{baseFunctionClass{ast.Lpad, 3, 3}},
+	ast.LTrim:      &lTrimFunctionClass{baseFunctionClass{ast.LTrim, 1, 1}},
+	ast.Mid:        &substringFunctionClass{baseFunctionClass{ast.Mid, 3, 3}},
+	ast.MakeSet:    &makeSetFunctionClass{baseFunctionClass{ast.MakeSet, 2, -1}},
+	ast.Oct:        &octFunctionClass{baseFunctionClass{ast.Oct, 1, 1}},
+	//This function is a synonym for LENGTH().
+	//See https://dev.mysql.com/doc/refman/5.7/en/string-functions.html#function_octet-length
```
Comment (lzmhhh123) [inline expression/builtin.go]: It's better not to comment. We could keep the format of the string functions without it.

Comment (gaoxingliang) [reply expression/builtin.go]: how about to put it at the end of line?  the go fmt tool did this auto format.

Comment (lzmhhh123) [reply expression/builtin.go]: You can simply delete the comment.

Comment (gaoxingliang) [reply expression/builtin.go]: aha.I think both is okay. and I fixed

**Pull request merged**